cosmo_tester/test_suites/image_based_tests/cfy_manager_test.py

```diff
@@ -98,6 +120,31 @@ def test_cfy_manager_configure(image_based_manager, logger, test_config):
         )
 
 
+def test_cfy_manager_upgrade(manager_5_1_0, ssh_key, logger, test_config):
+    example = get_example_deployment(
+        manager_5_1_0, ssh_key, logger, 'manager_upgrade', test_config)
+    example.upload_and_verify_install()
+    validate_agents(manager_5_1_0, example.tenant)
+
+    logger.info('Installing new RPM')
+    manager_5_1_0.run_command('yum install -y {rpm}'.format(
+        rpm=test_config['upgrade']['upgrade_rpm_path']), use_sudo=True)
+    logger.info('Upgrading manager')
+    manager_5_1_0.run_command('cfy_manager upgrade -v')
+
+    _validate_manager_status(manager_5_1_0)
+    validate_agents(manager_5_1_0, example.tenant)
+
+
+@retry(stop_max_attempt_number=30, wait_fixed=2000)
+def _validate_manager_status(manager):
```

Comment:
It'd probably be better to factor '_wait_for_healthy_broker_cluster out from full_cluster_test into utils and use it here as well.

Reply:
In fact, given that you're asserting on agent state as well, maybe it'd be best to factor out _validate_cluster_and_agents to utils as well and use it instead?

Reply:
I absolutely think this is a better idea, and I actually thought about it too. The only thing is that I don't see any utils file that is used across test_suits, so I thought there is a reason for it. Should I add one?

Reply:
Hmm... there should be stuff in the framework folder I think? If not, yes, make one.

Reply:
Cool, I will.